src/btree/bt_page.c

```diff
@@ -686,6 +686,7 @@ __inmem_row_leaf(WT_SESSION_IMPL *session, WT_PAGE *page)
         upd->durable_ts = unpack.tw.durable_start_ts;
         upd->start_ts = unpack.tw.start_ts;
         upd->txnid = unpack.tw.start_txn;
+        F_SET(upd, WT_UPDATE_PREPARE_RESTORED_FROM_DS);
```

Comment:
Why we only set the flag here？

Reply:
In all the below code flows, we are setting this flag. I moved it here instead of setting it everywhere.